daemon/response.go

```diff
@@ -281,55 +282,90 @@ func (f fileResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 //
 // Tip: “jq” knows how to read this; “jq --seq” both reads and writes this.
 type journalLineReaderSeqResponse struct {
-	io.ReadCloser
-	follow bool
+	readers []io.ReadCloser
+	follow  bool
+}
+
+// readAllLogsSorted combines the output of multiple io.ReadCloser's into a single
+// list of systemd.Log objects. The final list of systemd.Log objects will be sorted by
+// their timestamps, and this function will also close all the readers after reading
+// their logs.
+func (rr *journalLineReaderSeqResponse) readAllLogsSorted() ([]systemd.Log, error) {
+	var logs []systemd.Log
+	for _, r := range rr.readers {
+		dec := json.NewDecoder(r)
+		for {
+			var log systemd.Log
+			if err := dec.Decode(&log); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return nil, fmt.Errorf("cannot decode journal log: %v", err)
+			}
+			logs = append(logs, log)
+		}
+		r.Close()
```

Comment:
(nitpick^2) r.Close() could return an error but it seems this code is just moved around so not a concern here.